compaction/compaction_manager.hh

```diff
@@ -482,6 +484,11 @@ public:
     friend class compaction_manager_test;
 };
 
+namespace db { class config; }
+namespace replica { class database_config; }
+
+compaction_manager::config make_compaction_manager_config(const db::config& cfg, replica::database_config& dbcfg);
+
```

Comment:
looks like this function could be declared in replica/database.hh instead, so we don't have to forward declare those things from db and replica namespaces.

Reply:
what if we expand the list of params in make_compaction_manager_config? we want to stop including replica/database.hh from compaction_manager.cc. if we're declaring make_compaction_manager_config() in compaction_manager.hh, so its definition should be moved to compaction_manager.cc.

Reply:
but we can't because there's updatable static shares :-( we'll need to think about something.

Reply:
Maybe I'll just open code it in main and cql_test_env (as I did in an earlier draft that I didn't publish).
This will duplicate the code in both places but the current way it's done is awkward and really our of scope of both replica::database and compaction_manager.  The config stuff belongs to the application layer.

Reply:
If you open-code it into main.cc, it will be "duplicated" only between main and cql_test_env -- the same way as pretty much every other service config is (also mind #2705), so I'm for open-coding this helper

Reply:
Right.  We want to unify initialization anyhow.
Seems like you mentioned the wrong issue though... You probably meant #2795